Mobile Buy SDK/Mobile Buy SDK/Data/BUYClient+Customers.m

```diff
@@ -63,64 +65,64 @@ - (BUYRequestOperation *)createCustomerWithCredentials:(BUYAccountCredentials *)
 	}];
 }
 
-- (BUYRequestOperation *)createTokenForCustomerWithCredentials:(BUYAccountCredentials *)credentials customerJSON:(NSDictionary *)customerJSON callback:(BUYDataCustomerTokenBlock)block
+- (BUYRequestOperation *)activateCustomerWithCredentials:(BUYAccountCredentials *)credentials customerID:(NSString *)customerID token:(NSString *)token callback:(BUYDataCustomerTokenBlock)block
 {
-	NSURL *route = [self urlForCustomersToken];
-	return [self postRequestForURL:route object:credentials.JSONRepresentation completionHandler:^(NSDictionary *json, NSURLResponse *response, NSError *error) {
-		if (json && !error) {
-			BUYAuthenticatedResponse *authenticatedResponse = [BUYAuthenticatedResponse responseWithJSON:json];
-			self.customerToken = authenticatedResponse.accessToken;
-			
-			if (!customerJSON) {
-				[self getCustomerWithID:authenticatedResponse.customerID callback:^(BUYCustomer *customer, NSError *error) {
-					block(customer, self.customerToken, error);
-				}];
-			}
-			else {
-				BUYCustomer *customer = [self.modelManager customerWithJSONDictionary:json[@"customer"]];
-				block(customer, self.customerToken, error);
-			}
+	NSURL *url = [self urlForCustomersActivationWithID:customerID parameters:@{ @"token": token }];
+	
+	return [self putRequestForURL:url object:credentials.JSONRepresentation completionHandler:^(NSDictionary *json, NSHTTPURLResponse *response, NSError *error) {
+		NSString *email = json[@"customer"][@"email"];
+		if (email && !error) {
+			BUYAccountCredentialItem *emailItem = [BUYAccountCredentialItem itemWithEmail:email];
+			[self loginCustomerWithCredentials:[credentials credentialsByAddingItems:@[emailItem]] callback:block];
 		}
 		else {
 			block(nil, nil, error);
 		}
 	}];
 }
 
-- (BUYRequestOperation *)loginCustomerWithCredentials:(BUYAccountCredentials *)credentials callback:(BUYDataCustomerTokenBlock)block
+- (BUYRequestOperation *)updateCustomerWithCredentials:(BUYAccountCredentials *)credentials customerID:(NSString *)customerID callback:(BUYDataCustomerBlock)block
```

Comment:
If I understand this, this allows the customer to change their first name, last name, email, or password. Seems reasonable, although odd to use credentials instead of just using the customer object. But I suppose since most fields on customer are not editable, this is sensible.